scripts/operations/pyscripts/pyscripts/core/ssh/ssh_host.py

```diff
@@ -76,53 +77,85 @@ def get_password(self, reset_cached = False):
+        Gets the full domain name of this host using self.domain_suffix
+        """
         if self.domain_suffix:
             return "{}.{}".format(self.hostname, self.domain_suffix)
         else:
             return self.hostname
 
+    def get_ssh_command_to_connect_to_self(self):
+        """
+        Gets the SSH command to connect to this host. This assumes no underlying SSH connection.
+        """
+        return "ssh -o LogLevel=error -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null %s@%s" % (self.username, self.get_full_domain_name())
+
+    def get_ssh_command_to_connect_to_target(self, target_ssh_host):
+        """
+        Gets the SSH command to connect to the target host. This assumes this host is the underlying SSH connection.
+        """
+        if self.is_switch():
+            uses_vrf_Customer = "can" in target_ssh_host.domain_suffix or "chn" in target_ssh_host.domain_suffix or "cmn" in target_ssh_host.domain_suffix
+
+            vrf = " vrf Customer" if uses_vrf_Customer else ""
+
+            if self.is_mellanox_switch():
+                return "slogin%s %s@%s" % (vrf, target_ssh_host.username, self.get_target_hostname(target_ssh_host))
+            else:
+                return "ssh %s@%s%s" % (target_ssh_host.username, self.get_target_hostname(target_ssh_host), vrf)
+
+        else:
+            return "ssh -o LogLevel=error -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null %s@%s" % (target_ssh_host.username, self.get_target_hostname(target_ssh_host))
+
     def get_target_hostname(self, target_ssh_host):
         """
         Normalizes the hostname of a target host to connect to by:
 
-        1. if using nmnlb or hmnlb, then always goes to api.nmnlb.<system-domain> and api.hmnlb.<system-domain> respectively.
+        1. When using a switch and targetting another switch via the cmn network, uses the IP instead of the target switch instead of its hostname
 
-        2. When using vrf (i.e., this host is a switch) and going to cmn network, uses the IP instead.
+        It is so we can get around the bug https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1599. When we resolve that bug, then remove this workaround.
 
-        (2) is a workaround that substitutes the IP of the target host instead of the hostname to connect.
-        It is so we can get around the bug https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1600
-
-        Once that bug is resolved, either this entire method and the logic to use this method in ssh_connection should be removed,
-        or everything except the last line in this method that returns target_ssh_host.hostname should be removed.
+        2. If using nmnlb or hmnlb, then always goes to api.nmnlb.<system-domain> and api.hmnlb.<system-domain> respectively.
         """
-        if self.vrf and "cmn." in target_ssh_host.domain_suffix and "switch" in target_ssh_host.type:
+        if self.is_aruba_switch() and "cmn." in target_ssh_host.domain_suffix:
```

Comment:
I think this is okay for now but I think we could hit a case where the domain suffix is something like `can.eniacmn.com`.   That would falsely match the "cmn." case.

Reply:
Yes totally agree with you. At the very least I can split by period and compare. Will make the change.